File: Axiom/Internal/Ast/Literal.cs
namespace Axiom.Internal.Ast
{
    internal class Literal : Expression
    {
        internal enum LiteralType
        {
            Integer,
            Decimal,
            String,
            Nil,
            Logical
        }

        public Literal(object value, LiteralType type)
        {
            Value = value;
            Type = type;
        }

        public override void Accept(IVisitor visitor)
        {
            visitor.Visit(this);
        }

        public object Value { get; private set; }
        public LiteralType Type { get; private set; }
    }
}

File: src/main/java/com/moybl/axiom/ast/Literal.java
package com.moybl.axiom.ast;

import com.moybl.axiom.Position;

public class Literal extends Expression {

	public enum Kind {
		INTEGER,
		FLOAT,
		STRING,
		NIL,
		BOOLEAN
	}

	private Kind kind;
	private Object value;

	public Literal(Position position, Kind kind, Object value) {
		super(position);
		this.kind = kind;
		this.value = value;
	}

	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

	public Kind getKind() {
		return kind;
	}

	public Object getValue() {
		return value;
	}

}
